upup/pkg/fi/cloudup/awstasks/subnet.go

```diff
@@ -196,6 +205,22 @@ func (_ *Subnet) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Subnet) error {
 		}
 	}
 
+	if e.IPv6CIDR != nil && strings.HasPrefix(aws.StringValue(e.IPv6CIDR), "/") {
+		vpcIPv6CIDR, err := findVPCIPv6CIDR(t.Cloud, e.VPC.ID)
```

Comment:
Why limit this to Amazon-provided IPv6 associations? Why not just use e.VPC.IPv6CIDR?

Reply:
I cannot test with customer owned CIDRs. The AWS docs regarding IPv6 are not exactly clear either.
So, for now, this is meant to only support AWS owned CIDRs, where I know that can be exactly 1 per subnet with a /64 size and from exactly 1 bloc of /56 size.

Reply:
If there's a customer-specified `VPC.IPv6CIDR`, the intended behavior is obvious: take the slice off of the specified VPC IPv6CIDR.

Reply:
You could test using a shared VPC with a site-local IPv6 CIDR association.

It make sense to support only AWS-provided CIDRs in managed VPCs, especially for egress. But we shouldn't design out the shared VPC case. There's already code that pulls from `VPC.IPv6CIDR`; we should be consistent about using that when it's available.